mmv1/third_party/terraform/services/bigtable/duration_parsing_helper.go

```diff
@@ -0,0 +1,221 @@
+package bigtable
+
+import (
+	"errors"
+	"time"
+)
+
+var unitMap = map[string]uint64{
+	"ns": uint64(time.Nanosecond),
+	"us": uint64(time.Microsecond),
+	"µs": uint64(time.Microsecond), // U+00B5 = micro symbol
+	"μs": uint64(time.Microsecond), // U+03BC = Greek letter mu
+	"ms": uint64(time.Millisecond),
+	"s":  uint64(time.Second),
+	"m":  uint64(time.Minute),
+	"h":  uint64(time.Hour),
+	"d":  uint64(time.Hour) * 24,
+	"w":  uint64(time.Hour) * 24 * 7,
+}
+
+// ParseDuration parses a duration string.
+// A duration string is a possibly signed sequence of
+// decimal numbers, each with optional fraction and a unit suffix,
+// such as "300ms", "-1.5h" or "2h45m".
+// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+func ParseDuration(s string) (time.Duration, error) {
```

Comment:
Could you elaborate on why we need a custom function here? And could you add thorough unit tests for this function?

Reply:
We need it because we need support for "d" and "w", and the one from "time" isn't supporting it. It will never be supported as per https://github.com/golang/go/issues/17767.
I copied it from google3, copied tests too :)

Reply:
thanks!